server/chat.js

```diff
@@ -1520,6 +1520,16 @@ Chat.loadPlugins = function () {
 		if (plugin.loginfilter) Chat.loginfilters.push(plugin.loginfilter);
 		if (plugin.nicknamefilter) Chat.nicknamefilters.push(plugin.nicknamefilter);
 	}
+
+	let WL_FILES = FS('server/wavelength/chat-plugins/').readdirSync();
+	WL_FILES = WL_FILES.filter(file => file !== 'info.js');
```

Comment:
By the way, why is this here? I'm assuming it's just a remnant of the previous code in which `chat-plugins/info.js` was filtered and shifted to the beginning of the file queue so that it could be loaded first before any other plugin. I don't think this is necessary for WL plugins.

Honestly, Wavelength plugins should probably just be concatenated to the previous `files` variable and have them all load at the same time. Something like:

```js
let files = FS('server/chat-plugins/').readdirSync();
files = files.filter(file => file !== 'info.js');
files.unshift('info.js');

let wavelengthFiles = FS('server/wavelength/chat-plugins/').readdirSync();
files.push(...wavelengthFiles);
```
However, this is beyond the scope of this PR and probably should wait until the decision on the filesystem restructure has been decided upon.

Reply:
https://github.com/HoeenCoder/Wavelength/blob/4dcae63e82d155ee86ab3ce6cb3aa3228326205f/server/chat.js#L1510

Resolving the path might be an issue, any suggestions for identifying when the WL plugins begin within the loop?

The PRs dependent on this, since the path we're using is `wavelength/chat-plugins`

Reply:
`const plugin = require(`.${wavelengthFiles.includes(file) ? './wavelength' : ''}/chat-plugins/${file}`);`

Would this suffice?

Reply:
That's true, and it's something I should have thought about.
Well...you could technically check whether or not `file` exists within `wavelengthFiles` like this:
```js
const isWavelength = wavelengthFiles.includes(file);
const plugin = require(`./${wavelengthFile ? 'wavelength/' : ''}chat-plugins/${file}`);
```
However, whilst this ideally shouldn't result in negative behaviour, it could - for example, if both `files` and `wavelengthFiles` had the same filename `test.js`, then it would prefer to load from the Wavelength folder and not at all the main chat-plugins/ folder. The problem with this is that instead of overriding commands defined in the `chat-plugins/test.js` file, it will only load commands from the `wavelength/chat-plugins/test.js` file.
I don't like that behaviour. I think it would be best to just include the code where it checks for namefilters, nicknamefilters, et cetera like the main plugins do.

Reply:
You commented again before I managed to fully write my response, lol. Yeah, that proposal is pretty much identical to mine (so it's good to know that we think pretty similarly about things like that), but that still comes with the same caveat I mentioned in the response.

Reply:
introduce a new naming convention? wavelength plugins could have the prefix `wavelength-`? Thoughts?

Reply:
Well, if they're in a separate folder, then prefixing the files with `wavelength-` would be redundant. Just load them separately as they are now, but enable the Wavelength plugins to have the namefilters/nicknamefilters/et cetera that the main plugins can have.

Reply:
I would consider the prefix if we just moved all of the plugins into the same folder; however, I don't think that's the way to go. I like keeping them separated.